app/models/concerns/itemizable.rb

```diff
@@ -102,14 +102,21 @@ def total_quantity
     line_items.total
   end
 
-  def to_a
+  def line_item_hashes
     line_items.map do |l|
       # When the item isn't found, it's probably just inactive. This ensures it's available.
       item = Item.find(l.item_id)
       { item_id: item.id, name: item.name, quantity: l.quantity, active: item.active }.with_indifferent_access
     end
   end
 
+  # TODO: I'm not sure if this should just be removed or leave as a warning for now?
+  def to_a
+    Rails.logger.error("Calling Itemizable#to_a is deprecated. Use Itemizable#line_item_hashes instead.")
+    Rails.logger.error("Called on #{inspect} by #{caller}")
+    line_item_hashes
+  end
```

Comment:
@cielf my worry is that if we leave these as logs we're going to forget to check them. It's a bit more daring, but what would you think to put this behind a feature flag? If it's on, we error out, otherwise we go back to the old logic. Might be a bit more painful in the short term (we'd probably have to flip it a few times till we iron out the remaining uses), but we'd be a lot more sure that we're catching all the edge cases.

Reply:
Well,  I *am* looking at bugsnag at least once a week, already  and I'm interested in keeping the number of things we have to look at down.    What's the downside?

Reply:
The downside is that we might have some crashes that we have to look into / flip the flag back and it may inconvenience the impacted users until we do that.

Reply:
I just did a search on "to_a", albeit on another branch, and there's definitely under 100, the vast majority of which are obvious to someone who is really familiar with the system.      If we do a pass through those to catch anything less obvious,  I'm reasonably confident we'll be ok, and would be fine with the feature flag solution.

Reply:
The main problem is that `to_a` is **implicit** in a lot of different cases. For example, when calling `map` on something, it'll automatically call `to_a` on it if it's defined. That's my main worry here.

Reply:
Ok -- on balance, I'm for the feature flag and raise an error strategy.   It has definite advantages -- the biggest being that any problems will not be hidden,  *and* that we return the functionality to the users very quickly if we did miss something.     If we do the log,  and there's something that doesn't happen for a month, I feel we'll have a lot more trouble tracking it down.    There's  more of a chance of hard-to-track side effects from a solution that dies "silently" as far as the user experience is concerned.

Reply:
Flag added.